src/main/scala/squid/ir/Lowering.scala

```diff
@@ -41,7 +41,12 @@ trait Lowering extends Transformer {
               val ruh = RuntimeUniverseHelpers
               val typ = r.typ.typeArgs.last // TODO be careful that a method implemented without parameter list can implement a method with an empty list (eg toString), which can make this crash
               base.rep(MethodApp(r, ruh.FunctionType.symbol(reps.size).toType member ruh.sru.TermName("apply") asMethod, Nil, Args(reps:_*)::Nil, typ))
-            case (r, args) => lastWords(s"Not supported yet: vararg $args") // TODO
+            case (r, ArgsVarargs(Args(reps @ _*), Args(vreps @ _*))) =>
+              val ruh = RuntimeUniverseHelpers
+              val functionArity = reps.size + vreps.size
+              val typ = r.typ.typeArgs.last
+              val varargsAsSeq = base.rep(MethodApp(base.rep(StaticModule("scala.collection.Seq")), ruh.FunctionType.symbol(vreps.size).toType member ruh.sru.TermName("apply") asMethod, Nil, Args(vreps:_*)::Nil, ruh.SequenceType.apply(typ)))
```

Comment:
It looks like you're calling `FunctionN`'s `apply` method on object `Seq` here! It's wrong, as `Seq` has its own `apply` method (inherited from `scala.collection.generic.GenericCompanion `).
There is actually no need for messing with runtime universes here. You can see [in this code](https://github.com/epfldata/squid/blob/master/src/main/scala/squid/ir/AST.scala#L214-L221) a working example of building a `Seq` from a list of arguments ;^)